addon/adapters/osf-adapter.js

```diff
@@ -125,7 +125,7 @@ export default DS.JSONAPIAdapter.extend(HasManyQuery.RESTAdapterMixin, GenericDa
             var relatedType = singularize(snapshot.record[relationship].meta().type);
             res.data.forEach(item => {
                 var record = store.push(store.normalize(relatedType, item));
-                snapshot.record.resolveRelationship(relationship).addCanonicalRecord(record._internalModel.createSnapshot());
+                snapshot.record.resolveRelationship(relationship).addCanonicalRecord(record);
```

Comment:
I think the old version only worked by luck. Looking [at the source](https://github.com/emberjs/data/blob/master/addon/-private/system/relationships/state/relationship.js#L66) it seems a plain record and not a snapshot is needed here.